nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-2-6-processors/src/test/java/org/apache/nifi/processors/kafka/pubsub/TestPublishKafkaMock.java

```diff
@@ -368,7 +511,7 @@ private PublisherLease getPublisherLease(final Collection<ProducerRecord<byte[],
                 true,
                 patternAttributeName,
                 UTF_8,
-                null,
+                publishStrategy,
```

Comment:
In testing the code change, discovered this unit test was ignoring the specified publish strategy.  So several modified test cases in addition to the new one.